### PR TITLE
[release-v0.19] makefile: fix: Update controller-gen version to v0.14.0

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
The previous versions panic when running `make generate`.

This commit also fixes `check_commited_files` workflow.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -36,6 +36,13 @@ jobs:
   check_commited_files:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - name: Install Golang 1.20
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.20'
```
Comment (0xFelix) [inline .github/workflows/pull-request.yaml:42]: Is it possible to fetch that version from `go.mod`?

Comment (akrejcir) [reply .github/workflows/pull-request.yaml:42]: I'm not sure. If yes, we should first do it in `main` branch.

Comment (0xFelix) [reply .github/workflows/pull-request.yaml:42]: We do

https://github.com/kubevirt/ssp-operator/blob/5abe053dec0f255d6c59ebef3407ddb03da529f8/.github/workflows/pull-request.yaml#L41

Comment (akrejcir) [reply .github/workflows/pull-request.yaml:42]: Right. Sorry I forgot about that.
I will update this PR.

**Review by 0xFelix — COMMENTED** [review]
/approve

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -8,12 +8,12 @@ jobs:
   validate_metrics_docs:
     runs-on: ubuntu-latest
     steps:
-      - name: Install Golang 1.20
-        uses: actions/setup-go@v2
-        with:
-          go-version: '1.20'
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (dominikholler) [inline .github/workflows/pull-request.yaml:12]: Will renovate update such lines in the future, e.g. like in https://github.com/dominikholler/webdavserver/pull/5 ?

Comment (akrejcir) [reply .github/workflows/pull-request.yaml:12]: Yes, it will.

**Comment by akrejcir** [comment]
Overriding CI, because the changes in this PR are not tested by functional tests.

/override ci/prow/e2e-functests

**Pull request merged**